### DISCUSSION TITLE
Address Range, setting and using

### DISCUSSION THREAD

**Discussion by karel201712** [memtest86plus/memtest86plus#393] [Q&A]
Hi!
I have 64GB of RAM (4x16GB). It take a very long continuous block of time to test it all with reasonable number of passes. My idea is to specify four address ranges and test them separately, which should allow for smaller chunks of time needed at a time, which is an issue for me.

Is this a good idea and if not, may I ask why (not to dispute, but to learn)?

If my idea (see above) is OK, pls, let me ask a very silly sounding question: to achieve the goal (see above), I would set lower/upper limits like this, right?
0G - 16G
16G - 32G
32G - 48G
48G - 65.7G
(By default or after using "Test all memory" Memtest86+ displays "Current range: 0KB - 65.7GB")

And perhaps the last question: I have noticed that when I set the lower/upper limits, the input accepts, apart from numbers, the following letters T, P, G, K, x (not capitalized), M. I'd assume T=TB, P=PB, G=GB, K=kB and M=MB, but what about "x"? Some kind of a useful wildcard perhaps?

Many thanks in advance for any kind help!
:)
K

**Comment by martinwhitaker** [comment]
It is best to run the address tests over the full range of memory, as they check that there are no stuck address bits. Otherwise I don't think it matters. But if you are planning to do a reasonable number of passes, I can't see that you would gain much, e.g. 4 separate runs doing 4 passes over the full range should take a similar time to 4 separate runs, each doing 16 passes on a quarter of the range (it won't be quite the same because the first pass of each run is shorter).

The number input format is explained in lib/read.h. We should also put that in the README.

**Reply by karel201712** [reply]
if anybody is looking for the above mentioned file, it can be found here:
https://github.com/memtest86plus/memtest86plus/blob/main/lib/read.h

**Reply by karel201712** [reply]
> (...) e.g. 4 separate runs doing 4 passes...

If I understand you right, then you are suggesting something I did not consider. Let me see if I get it...
Are you saying that from the two example below, the first one is better for catching errors than the second one?

EXAMPLE 1:
4 separate runs, each over full address range, each run with 2-3 passes, (these 4 runs would be spread over cca two weeks)

EXAMPLE 2:
over cca a month do the following series four times:
run over 1st quarter of the full address range with 8-12 passes
run over 2nd quarter of the full address range with 8-12 passes
run over 3rd quarter of the full address range with 8-12 passes
run over 4th quarter of the full address range with 8-12 passes

:)
K

**Reply by martinwhitaker** [reply]
No, I'm saying that

4 runs of 4 passes over the full range

should take roughly the same time and provide roughly the same coverage as

1 run of 16 passes over the 1st quarter
1 run of 16 passes over the 2nd quarter
1 run of 16 passes over the 3rd quarter
1 run of 16 passes over the 4th quarter

except that the second option won't catch an address bit fault in the top two address bits.

Actually, because the first pass of a run is shorter, it might be a closer match if the first option was 4 runs of 5 passes.

**Reply by karel201712** [reply]
thank you for your kind reply! sadly, I'm not sure what to make of it.

what is the significant difference between my suggestion in my previous post (https://github.com/memtest86plus/memtest86plus/discussions/393#discussioncomment-10037773) and your description in your latest post (https://github.com/memtest86plus/memtest86plus/discussions/393#discussioncomment-10037884), pls?

**Comment by debrouxl** [comment]
Let's add that shorter runs can't be better for detecting longer-term stability issues, due to e.g. thermal effects.
